heudiconv/dicoms.py

```diff
@@ -314,19 +315,55 @@ def group_dicoms_into_seqinfos(files, grouping, file_filter=None,
     return seqinfos
 
 
-def get_dicom_series_time(dicom_list):
-    """Get time in seconds since epoch from dicom series date and time
-    Primarily to be used for reproducible time stamping
+def get_timestamp_from_series(dicom_list: list[str]) -> int:
+    """try to return a timestamp indicating when the first dicom in dicom_list was created, or if that
+    info isn't present then a reproducible integer. This is used in setting mtimes reproducibly
+
+    Args:
+        dicom_list (list[str]): list of strings pointing to existing dicom files
+
+    Returns:
+        int: either an int representing when the first dicom was created, 
+            in number of seconds since epoch, or if no datetime info is found then a hash of the
+            SeriesInstanceUID (meaningless value, but reproducible)
+
     """
-    import time
     import calendar
 
     dicom = dcm.read_file(dicom_list[0], stop_before_pixels=True, force=True)
-    dcm_date = dicom.SeriesDate  # YYYYMMDD
-    dcm_time = dicom.SeriesTime  # HHMMSS.MICROSEC
-    dicom_time_str = dcm_date + dcm_time.split('.', 1)[0]  # YYYYMMDDHHMMSS
-    # convert to epoch
-    return calendar.timegm(time.strptime(dicom_time_str, '%Y%m%d%H%M%S'))
+    if (dicom_datetime := get_datetime_from_dcm(dicom)):
```

Comment:
`:=` was added in python 3.8. We still remain compatible with 3.7. 
```suggestion
    dicom_datetime = get_datetime_from_dcm(dicom)
    if dicom_datetime:
```

Reply:
The walrus operator has been removed

```
rg ":="  
utils/test-compare-two-versions.sh
16:outdir=${OUTDIR:=compare-versions}
```